doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -32,169 +32,456 @@
+      move_group_.getCurrentState()->getJointModelGroup(planning_group_);
 
   // We will use the :planning_scene_interface:`PlanningSceneInterface`
-  // class to add and remove collision objects in our "virtual world" scene
-  moveit::planning_interface::PlanningSceneInterface planning_scene_interface;
+  // class to add and remove collision objects in our "virtual world" scene.
+  moveit::planning_interface::PlanningSceneInterface planning_scene_interface_;
+  // These vectors hold additional information about the objects added to the "virtual world".
+  std::vector<moveit_msgs::CollisionObject> collision_objects_;
+  std::vector<std::string> object_ids_;
+
+  // The package :moveit_visual_tools:`MoveItVisualTools` provides many capabilties for visualizing objects, robots,
+  // and trajectories in RViz as well as debugging tools such as step-by-step introspection
+  // of a script.
+  moveit_visual_tools::MoveItVisualTools visual_tools_ = moveit_visual_tools::MoveItVisualTools("panda_link0");
+  // END_SUB_TUTORIAL
+
+  // A few other pieces of data are stored in the class
+  const std::string name_ = "tutorial";
+  Eigen::Isometry3d text_pose_ = Eigen::Isometry3d::Identity();
+};
 
-  // Raw pointers are frequently used to refer to the planning group for improved performance.
-  const robot_state::JointModelGroup* joint_model_group =
-      move_group.getCurrentState()->getJointModelGroup(PLANNING_GROUP);
+int main(int argc, char** argv)
+{
+  // ROS startup
+  ros::init(argc, argv, "move_group_interface_tutorial");
+  ros::NodeHandle node_handle;
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
 
-  // Visualization
-  // ^^^^^^^^^^^^^
-  //
-  // The package MoveItVisualTools provides many capabilties for visualizing objects, robots,
-  // and trajectories in RViz as well as debugging tools such as step-by-step introspection of a script
-  namespace rvt = rviz_visual_tools;
-  moveit_visual_tools::MoveItVisualTools visual_tools("panda_link0");
-  visual_tools.deleteAllMarkers();
-
-  // Remote control is an introspection tool that allows users to step through a high level script
-  // via buttons and keyboard shortcuts in RViz
-  visual_tools.loadRemoteControl();
-
-  // RViz provides many types of markers, in this demo we will use text, cylinders, and spheres
-  Eigen::Isometry3d text_pose = Eigen::Isometry3d::Identity();
-  text_pose.translation().z() = 1.75;
-  visual_tools.publishText(text_pose, "MoveGroupInterface Demo", rvt::WHITE, rvt::XLARGE);
-
-  // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations
-  visual_tools.trigger();
-
-  // Getting Basic Information
-  // ^^^^^^^^^^^^^^^^^^^^^^^^^
-  //
-  // We can print the name of the reference frame for this robot.
-  ROS_INFO_NAMED("tutorial", "Planning frame: %s", move_group.getPlanningFrame().c_str());
+  // Create and setup the MoveGroupInterfaceTutuorial instance
+  MoveGroupInterfaceTutorial move_group_interface_tutorial;
```

Comment:
I don't like this class. Its basically an interface for an interface to MoveGroup, which is a wrapper to MoveIt already... this only further complicates using MoveIt.